draft-ietf-taps-interface.md

```diff
@@ -258,6 +258,169 @@ Errors and other notifications also happen asynchronously on the Connection.
 through Actions and Events in each phase of a Connection, following the phases
 described in {{I-D.ietf-taps-arch}}.
 
+## Usage Examples
+
+The following usage examples illustrate how an application might use a
+Transport Services Interface to
+
+- Act as a server: Listen for incoming connections, receive a request, and send
+  a response, see {{server-example}}.
+- Act as a client: Connect to a remote endpoint, send a request, and receive a
+  response, see {{client-example}}.
+- Act as a peer: Connect to a remote endpoint using Rendezvous, send a Message,
+  and receive a Message, see {{peer-example}}.
+
+The examples in this section presume that a transport protocol is available
+between the endpoints which provides Reliable Data Transfer, Preservation of
+data ordering, and Preservation of Message Boundaries. In this case, the
+application can choose to receive only complete messages.
+
+If none of the available transport protocols provides Preservation of Message
+Boundaries, but there is a transport protocol which provides a reliable ordered
+octet stream, an application may receive this octet stream as partial
+Messages and transform it into application-layer Messages.  Alternatively,
+an application may provide a Deframer, which is a function that transforms an
+octet stream into a sequence of Messages, see {{receive-framing}}.
+
+
+### Server Example
+
+This is an example of how an application might listen for incoming Connections
+using the Transport Services Interface, receive a request, and send a response.
+
+~~~
+LocalSpecifier := NewLocalEndpoint()
+LocalSpecifier.WithInterface("en0")
```

Comment:
I don't like ever recommending using an interface by name; people often make incorrect assumptions about which interface is named which.

Reply:
Agreed, hardcoding these in an application might be a bad idea.
I took this example from the Transport Properties Section, so if we decide it's bad, maybe we should change it there as well.

What about: LocalSpecifier.WithInterface("any")? Or is there a better way to specify an interface to listen on than with interface names?